packages/abstraction/test/ContractGenerator.test.ts

```diff
@@ -1,3 +1,98 @@
+import path from 'path';
+import { ContractGenerator } from "../src";
+import { MultipleContractAdapter, EOSContractAdapter, ICONContractAdapter, ModelContractAdapter } from "@aloxide/bridge";
+import * as  SchemaValidator from '../src/SchemaValidator';
+
 describe('test ContractGenerator', () => {
-  it.todo('test contract generator');
+  let contractGenerator;
+  const logger = {
+    info: jest.fn(),
+    debug: jest.fn(),
+  };
+  const adapter = new MultipleContractAdapter();
+
+  adapter.addAdapters(
+    new EOSContractAdapter(),
+    new ICONContractAdapter(),
+    new ModelContractAdapter(),
+  );
+
+  const checkName = val => {
+    return /^[1-5a-zA-Z]+/.test(val)
+  }
+
+  const checkType = val => {
+    const supportedType = ["uint64_t", "number", "string", "array", "bool"];
+    return supportedType.indexOf(val) !== -1
+  }
+
+  const requiredSchema = {
+    entities: [{
+      name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+      fields: [{
+        name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+        type: { type: String, required: true, use: { checkType } },
+      }],
+      key: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } }
+    }
+    ]
+  }
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
```

Comment:
should add `restoreAllMocks` to jest config

Reply:
I already add it into jest.config.js file

Reply:
then this line is redundant